### ISSUE TITLE
[BUG] IOS notification

### ISSUE THREAD

**Issue by Marcoant007** [WhiskeySockets/Baileys#607]
**BUG**
When you scan the code and connect to IOS, you no longer receive notifications on your mobile screen.

**To Reproduce**
1. Created a new session qrcode
2. Scan the code in ios
3. No more notifications if someone sends you a message

**Expected behavior**

**Environment (please complete the following information):**
 - Is this on a server? Yes 
 - What do your `connectOptions` look like? 
 ```bash
 const handleConnectionUpdate = SSE ? handleSSEConnectionUpdate : handleNormalConnectionUpdate;
  const { state, saveCreds } = await useSession(sessionId);
  const socket = makeWASocket({
    version: [2,2323,4],
    printQRInTerminal: true,
    browser: Browsers.ubuntu('Chrome'),
    generateHighQualityLinkPreview: true,
    markOnlineOnConnect: false,
    ...socketConfig,
    auth: {
      creds: state.creds,
      keys: makeCacheableSignalKeyStore(state.keys, logger),
    },
    logger,
    shouldIgnoreJid: (jid) => isJidBroadcast(jid),
    getMessage: async (key) => {
      const data = await prisma.message.findFirst({
        where: { remoteJid: key.remoteJid!, id: key.id!, sessionId },
      });
      return (data?.message || undefined) as proto.IMessage | undefined;
    },
  });
 ```

**Additional context**
I tried using sock.sendPresenceUpdate('unavailable'), but that didn't solve the problem

**Comment by SuhailTechInfo** [comment]
> **BUG** When you scan the code and connect to IOS, you no longer receive notifications on your mobile screen.
> 
> **To Reproduce** Steps to reproduce the behavior:
> 
> 1. Created a new session qrcode
> 2. Scan the code in ios
> 3. No more notifications if someone sends you a message
> 
> **Expected behavior** A clear and concise description of what you expected to happen.
> 
> **Environment (please complete the following information):**
> 
> * Is this on a server? Yes
> * What do your `connectOptions` look like?
> 
> ```shell
> const handleConnectionUpdate = SSE ? handleSSEConnectionUpdate : handleNormalConnectionUpdate;
>  const { state, saveCreds } = await useSession(sessionId);
>  const socket = makeWASocket({
>    version: [2,2323,4],
>    printQRInTerminal: true,
>    browser: Browsers.ubuntu('Chrome'),
>    generateHighQualityLinkPreview: true,
>    markOnlineOnConnect: false,
>    ...socketConfig,
>    auth: {
>      creds: state.creds,
>      keys: makeCacheableSignalKeyStore(state.keys, logger),
>    },
>    logger,
>    shouldIgnoreJid: (jid) => isJidBroadcast(jid),
>    getMessage: async (key) => {
>      const data = await prisma.message.findFirst({
>        where: { remoteJid: key.remoteJid!, id: key.id!, sessionId },
>      });
>      return (data?.message || undefined) as proto.IMessage | undefined;
>    },
>  });
> ```
> 
> **Additional context** I tried using sock.sendPresenceUpdate('unavailable'), but that didn't solve the problem

***I think, That issue is not because of `sendPresenceUpdate('unavailable')` !***
<br>
### Maybe you have used below method:
- for auto read messages
```
     conn.readMessages([message.key])  // Comment this method
```


***In my case when i enable it, then bot auto read all messages, and am not receive any notifications on my mobile screen***

**Comment by kwangmeng** [comment]
> **BUG** When you scan the code and connect to IOS, you no longer receive notifications on your mobile screen.
> 
> **To Reproduce** Steps to reproduce the behavior:
> 
> 1. Created a new session qrcode
> 2. Scan the code in ios
> 3. No more notifications if someone sends you a message
> 
> **Expected behavior** A clear and concise description of what you expected to happen.
> 
> **Environment (please complete the following information):**
> 
> * Is this on a server? Yes
> * What do your `connectOptions` look like?
> 
> ```shell
> const handleConnectionUpdate = SSE ? handleSSEConnectionUpdate : handleNormalConnectionUpdate;
>  const { state, saveCreds } = await useSession(sessionId);
>  const socket = makeWASocket({
>    version: [2,2323,4],
>    printQRInTerminal: true,
>    browser: Browsers.ubuntu('Chrome'),
>    generateHighQualityLinkPreview: true,
>    markOnlineOnConnect: false,
>    ...socketConfig,
>    auth: {
>      creds: state.creds,
>      keys: makeCacheableSignalKeyStore(state.keys, logger),
>    },
>    logger,
>    shouldIgnoreJid: (jid) => isJidBroadcast(jid),
>    getMessage: async (key) => {
>      const data = await prisma.message.findFirst({
>        where: { remoteJid: key.remoteJid!, id: key.id!, sessionId },
>      });
>      return (data?.message || undefined) as proto.IMessage | undefined;
>    },
>  });
> ```
> 
> **Additional context** I tried using sock.sendPresenceUpdate('unavailable'), but that didn't solve the problem

Im having same issue as well, upon session connected, notifications are not sent to mobile app.

**Comment by brunocascio** [comment]
Same here, is there any workaround?

**Comment by diegoluan23** [comment]
+1

**Comment by Gutistavo** [comment]
+1

**Comment by josembelamendia** [comment]
any solutions to this? im having the same issue, no push notifications are shown when i connect the whatsapp with baileys.

**Comment by teren-papercutlabs** [comment]
@josembelamendia im hesitant to recommend this as a solution, but i have a cron job that sets presence to unavailable every 5 minutes and my customer says they get notifications now

```

cron.schedule('*/5 * '*, async () => {
    const logger = new Logger('PRESENCE_CRON');
    try {
      logger.info('Setting all instances to unavailable...');
      // Get all instances
      const instances = await prismaRepository.instance.findMany({
        where: { connectionStatus: 'open' }
      });
      for (const instance of instances) {
        try {
          const serverUrl = configService.get<HttpServer>('SERVER').URL;
          const apiKey = configService.get<Auth>('AUTHENTICATION').API_KEY.KEY;
          // URL encode the instance name for the API call
          const encodedName = encodeURIComponent(instance.name);
          await axios.post(
            ${serverUrl}/instance/setPresence/${encodedName},
            { presence: 'unavailable' },
            {
              headers: {
                'Content-Type': 'application/json',
                'apikey': apiKey
              }
            }
          );
          logger.info(Set presence unavailable for instance:
  ${instance.name});
        } catch (error) {
          logger.error(Failed to set presence for ${instance.name}:
  ${error.message});
        }
      }
    } catch (error) {
      logger.error(Presence cron job failed: ${error.message});
    }
  });

```
you might want to try it but it's ugly

**Comment by josembelamendia** [comment]
Yeah, it's working with something like that, thank you so much 🔼🔼
